src/lib.rs

```diff
@@ -76,6 +76,24 @@ extern "C" {
     pub fn TYPEOF(x: SEXP) -> SEXPTYPE;
 }
 
+impl From<Rboolean> for bool {
+    fn from(value: Rboolean) -> Self {
+        match value {
+            Rboolean::FALSE => false,
+            Rboolean::TRUE => true,
+        }
+    }
+}
```

Comment:
Can you guarantee that there will not be any other case for `Rboolean` enum? From definition perspective it is exhaustive, but can `R` pass us some garbage to treat as `TRUE`?

Reply:
R does pass us garbage to interpret as TRUE. I believe I wrote about it in the discord.. It is related to whether something is an S4 object. There, we get 16 and not 1.
But here, this is a conversion between rust's boolean and this representation for Rboolean. We don't have the u32-conversion here. If that conversion was present (which I may put in a PR going forward), then it would be `From<u32>`and it will have to interpret `>0` as `Rboolean::TRUE`.

Reply:
And I can't cast a u32 to a Rboolean at the moment.

Reply:
@Ilia-Kosenkov what do you think?

Reply:
What I meant is, R deals in `u32` (or `i32` or whatever). Then it has constants defined, like `FALSE` and `TRUE`. Based on that we generated an exhaustive enum. But if there will be a case that some function (due to binding generation) returns value other than `1` or `0` which is stored as `Rboolean`, then this will likely panic badly. If there are cases where we have functions in `libR-sys` that return `Rboolean` but can smuggle in `16` or any other value, we need to make this enum non-exhaustive and explicitly handle the case when it is neither `TRUE` or `FALSE`.

If however, this may never happen, then I am ok with current solution.

Reply:
```rust
  pub fn Rf_isS4_old(arg1: SEXP) -> i32;
```
Does it return `i32` or `Rboolean` in `libR-sys`?

Reply:
```rust
 pub fn Rf_isS4(arg1: SEXP) -> Rboolean;
```
This is your problem. If `Rboolean` is an enum of two values 
```rust
#[repr(u32)]
#[non_exhaustive]
#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
pub enum Rboolean {
    #[doc = ", MAYBE"]
    FALSE = 0,
    #[doc = ", MAYBE"]
    TRUE = 1,
}
```
but this function return `16u32`, Rust will likely explode with UB. You will write exhaustive `match` where non of branches (`TRUE` or `FALSE`) will work.

Reply:
E.g. this https://github.com/rust-lang/rust-memory-model/issues/41

Reply:
When I tried to make the example into a test, I got weird errors. The discriminant value is used in the hashing, and it is different, but I can't see it used in the partialeq, yet it still affect it.

Reply:
Please check what I've pushed now.. Otherwise, I'm calling it. I believe it would be a major disservice to not have `From<bool>`, especially since it is littered everywhere in the API.

We should just keep an eye on these things if they break. I already went through the remainder of the API, and only know of this one silly instance.